src/entities/plugin.ts

```diff
@@ -0,0 +1,27 @@
+// The plugin entity
+export interface Plugin {
```

Comment:
I think the naming convention would be a bit nicer to use if we called the interfaces `PluginInterface` here and in the class definition had:
```export class Plugin implements PluginInterface```
That way we keep the names of the objects to be just the object itself.

Reply:
that's fine, but as you know java convention is the Impl way, but will update